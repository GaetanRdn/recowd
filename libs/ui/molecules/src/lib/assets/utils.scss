@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
