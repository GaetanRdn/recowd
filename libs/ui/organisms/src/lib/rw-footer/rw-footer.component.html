<footer>
  <rc-toolbar>
    <rc-nx-custom-icon [width]="24"></rc-nx-custom-icon>
    <rc-angular-custom-icon [width]="24"></rc-angular-custom-icon>
    <rc-jest-custom-icon [width]="24"></rc-jest-custom-icon>
    <rc-storybook-custom-icon [width]="24"></rc-storybook-custom-icon>
    <rc-sass-custom-icon [width]="24"></rc-sass-custom-icon>
    <rc-cypress-custom-icon [width]="24"></rc-cypress-custom-icon>
  </rc-toolbar>
</footer>
