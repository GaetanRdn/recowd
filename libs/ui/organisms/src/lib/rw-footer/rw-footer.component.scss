:host {
  rc-toolbar {
    gap: 1rem;
    justify-content: center;
  }
}
