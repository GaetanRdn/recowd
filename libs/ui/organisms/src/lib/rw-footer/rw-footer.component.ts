import { ChangeDetectionStrategy, Component } from '@angular/core';
import { ToolbarComponent } from '@recowd/ui-atoms';
import {
  AngularCustomIconComponent,
  CypressCustomIconComponent,
  JestCustomIconComponent,
  NxCustomIconComponent,
  SassCustomIconComponent,
  StorybookCustomIconComponent,
} from '@recowd/ui/molecules';

@Component({
  selector: 'rc-rw-footer',
  standalone: true,
  imports: [
    ToolbarComponent,
    NxCustomIconComponent,
    AngularCustomIconComponent,
    JestCustomIconComponent,
    CypressCustomIconComponent,
    SassCustomIconComponent,
    StorybookCustomIconComponent,
  ],
  templateUrl: './rw-footer.component.html',
  styleUrls: ['./rw-footer.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class RwFooterComponent {}
