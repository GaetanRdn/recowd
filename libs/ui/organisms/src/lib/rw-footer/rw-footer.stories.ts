import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { RwFooterComponent } from './rw-footer.component';

export default {
  title: 'organisms/rw-footer',
  decorators: [
    moduleMetadata({
      imports: [RwFooterComponent],
    }),
  ],
} as Meta<RwFooterComponent>;

const Template: Story<RwFooterComponent> = () => ({
  template: `<rc-rw-footer></rc-rw-footer>`,
});

export const Default = Template.bind({});
