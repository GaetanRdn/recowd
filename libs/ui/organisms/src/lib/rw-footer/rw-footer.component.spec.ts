import { TestBed, waitForAsync } from '@angular/core/testing';

import { RwFooterComponent } from './rw-footer.component';
import { TemplateLookup } from '@recowd/test/utils';

describe('RwFooterComponent', () => {
  let templateLookup: TemplateLookup<RwFooterComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      imports: [RwFooterComponent],
    }).compileComponents();

    templateLookup = new TemplateLookup(RwFooterComponent);
    templateLookup.detectChanges();
  }));

  it('should create', () =>
    expect(templateLookup.fixture.nativeElement.children).toMatchSnapshot());
});
