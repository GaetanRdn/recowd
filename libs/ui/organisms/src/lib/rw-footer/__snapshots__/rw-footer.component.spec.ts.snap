// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RwFooterComponent should create 1`] = `
HTMLCollection [
  <footer>
    <rc-toolbar>
      <rc-nx-custom-icon>
        <rc-custom-icon>
          <object
            data="assets/images/nx.svg"
            height="24"
            type="image/svg+xml"
            width="24"
          >
            <img
              alt="assets/images/nx"
              src="assets/images/nx.png"
            />
          </object>
        </rc-custom-icon>
      </rc-nx-custom-icon>
      <rc-angular-custom-icon>
        <rc-custom-icon>
          <object
            data="assets/images/angular.svg"
            height="24"
            type="image/svg+xml"
            width="24"
          >
            <img
              alt="assets/images/angular"
              src="assets/images/angular.png"
            />
          </object>
        </rc-custom-icon>
      </rc-angular-custom-icon>
      <rc-jest-custom-icon>
        <rc-custom-icon>
          <object
            data="assets/images/jest.svg"
            height="24"
            type="image/svg+xml"
            width="24"
          >
            <img
              alt="assets/images/jest"
              src="assets/images/jest.png"
            />
          </object>
        </rc-custom-icon>
      </rc-jest-custom-icon>
      <rc-storybook-custom-icon>
        <rc-custom-icon>
          <object
            data="assets/images/storybook.svg"
            height="24"
            type="image/svg+xml"
            width="24"
          >
            <img
              alt="assets/images/storybook"
              src="assets/images/storybook.png"
            />
          </object>
        </rc-custom-icon>
      </rc-storybook-custom-icon>
      <rc-sass-custom-icon>
        <rc-custom-icon>
          <object
            data="assets/images/sass.svg"
            height="24"
            type="image/svg+xml"
            width="24"
          >
            <img
              alt="assets/images/sass"
              src="assets/images/sass.png"
            />
          </object>
        </rc-custom-icon>
      </rc-sass-custom-icon>
      <rc-cypress-custom-icon>
        <rc-custom-icon>
          <object
            data="assets/images/cypress.svg"
            height="24"
            type="image/svg+xml"
            width="24"
          >
            <img
              alt="assets/images/cypress"
              src="assets/images/cypress.png"
            />
          </object>
        </rc-custom-icon>
      </rc-cypress-custom-icon>
    </rc-toolbar>
  </footer>,
]
`;
